data-service/src/main/java/uk/gov/gchq/palisade/service/data/model/AuditErrorMessage.java

```diff
@@ -31,59 +31,113 @@
 /**
  * Represents information for an error that has occurred during the processing of a request. This information is
  * forwarded to the audit-service.
- * Note all of the services can potentially send an error message.
  */
 public final class AuditErrorMessage extends AuditMessage {
 
     private final Throwable error;  //Error that occurred
 
     @JsonCreator
     private AuditErrorMessage(
+            final @JsonProperty("leafResourceId") String leafResourceId,
+            final @JsonProperty("token") String token,
             final @JsonProperty("userId") String userId,
             final @JsonProperty("resourceId") String resourceId,
-            final @JsonProperty("context") JsonNode context,
+            final @JsonProperty("context") Context context,
             final @JsonProperty("attributes") Map<String, Object> attributes,
             final @JsonProperty("error") Throwable error) {
 
-        super(userId, resourceId, context, attributes);
+        super(leafResourceId, token, userId, resourceId, context, attributes);
+
         this.error = Optional.ofNullable(error).orElseThrow(() -> new IllegalArgumentException("Error" + " cannot be null"));
+
     }
 
     @Generated
     public Throwable getError() {
         return error;
     }
 
+
     /**
-     * Builder class for the creation of instances of the AuditSuccessMessage.  This is a variant of the Fluent Builder
+     * Builder class for the creation of instances of the AuditErrorMessage.  This is a variant of the Fluent Builder
      * which will use Java Objects or JsonNodes equivalents for the components in the build.
      */
     public static class Builder {
         /**
          * Starter method for the Builder class.  This method is called to start the process of creating the
-         * AuditSuccessMessage class.
+         * AuditErrorMessage class.
          *
-         * @return interface {@link IUserId} for the next step in the build.
+         * @return interface {@link ILeafResourceId} for the next step in the build.
          */
-        public static IUserId create() {
-            return userId -> resourceId -> context -> attributes -> error ->
-                    new AuditErrorMessage(userId, resourceId, context, attributes, error);
+        public static ILeafResourceId create() {
```

Comment:
The AuditErrorMessage shouldnt contain the token object, if its needed, an object should be created that pairs a token with an AuditErrorMessage

Also revert the re-ordering of the object, userId should be first, like the other services AuditErrorMessage obejcts

Reply:
This AuditErrorMessage  is the exception to the rule.  The client will send a DataRequest that will have the token and the leafResourceId in the body of the message.  It is worth a mention that if the request fails in the first service call for authorising the request, these will be the only two parameters in the AuditErrorMessage.   As for the re-ordering it is in line with the convention that I understood we are using which is  The first in is the first in the set of parameters.  In the sequence that we have in the Data Service, the request comes in with leafResourceId and token.  The subsequent (successful) service call to authorise the request will provide the userID, resourceID, context and attributes.  If something goes wrong with the service call to read the resource, it will then add the error.    Key point here is that it is different to the other AuditErrorMessage and the same applies to AuditSuccessMessage.  Note I have raised a ticket related to this as the Audit Service will have to handle this variant.

Reply:
All services send the token to the error topic, but they do so via the Auditable***Response class which attaches the AuditError/SuccessMessage to a Response object, which contains the request information and the token. 

The order that the variables come in to the service doesnt change the order that they are entered into the Audit***Messge objects, they should be identical to the other services